src/lndk_offers.rs

```diff
@@ -44,9 +63,53 @@ pub fn decode(offer_str: String) -> Result<Offer, Bolt12ParseError> {
     offer_str.parse::<Offer>()
 }
 
+// request_invoice builds an invoice request and tries to send it to the offer maker.
+pub async fn request_invoice(
+    client: Client,
+    custom_msg_client: impl SendCustomMessage,
+    pubkey: PublicKey,
+    offer: Offer,
+    introduction_node: PublicKey,
+    blinded_path: BlindedPath,
+) -> Result<(), OfferError<Secp256k1Error>> {
+    let mut signer_clone = client.clone();
+    let node_client = signer_clone.signer();
+    let node_signer = LndNodeSigner::new(pubkey, node_client);
+    let messenger_utils = MessengerUtilities::new();
+    let onion_messenger = OnionMessenger::new(
+        &messenger_utils,
+        &node_signer,
+        &messenger_utils,
+        &DefaultMessageRouter {},
+        IgnoringMessageHandler {},
+        IgnoringMessageHandler {},
+    );
+
+    // Ensure that the introduction node we need to connect to actually supports onion messaging.
+    let onion_support =
+        lookup_onion_support(&introduction_node.clone(), client.clone().lightning()).await;
+    if !onion_support {
+        return Err(OfferError::SendError(SendError::InvalidFirstHop));
+    }
+
+    let invoice_request =
+        create_invoice_request(client.clone(), offer, vec![], Network::Regtest, 20000)
```

Comment:
Amount is hardcoded in here? 

Shouldn't we (in `create_invoice_request`):
* Allow the caller to specify `Some(u64)` to set an amount to pay
* If the offer has an amount, ensure that `Some(user amount)` is > offer amount
* If the offer does not have an amount (or is zero amt), ensure that we have `Some(user amount)`